homeassistant/components/weatherkit/coordinator.py

```diff
@@ -0,0 +1,75 @@
+"""DataUpdateCoordinator for WeatherKit integration. Updated every 15 minutes."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Literal
+
+from apple_weatherkit.client import (
+    WeatherKitApiClient,
+    WeatherKitApiClientAuthenticationError,
+    WeatherKitApiClientError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LATITUDE, CONF_LOCATION, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER
+
+
+class WeatherKitDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from the API."""
+
+    config_entry: ConfigEntry
+    supported_data_sets: list[
+        Literal["currentWeather", "forecastDaily", "forecastHourly"]
```

Comment:
Can these be made constants or moved to the library? (not sure whats best here)

Reply:
This is something I was torn about too. I think since that these values are closely related to the API details (they are the actual keys that get sent to WeatherKit), I can abstract it by creating an enum in the library for the possible data sets, and use that when passing data to and from the library. That way the client isn't concerned with the underlying values, and we have nice types.

Reply:
Sounds beautiful